src/main/java/jep/ClassList.java

```diff
@@ -95,16 +94,23 @@ private void loadClassPath() {
         while (!queue.isEmpty()) {
             String el = queue.remove();
 
-            if (!el.toLowerCase().endsWith(".jar")) {
-                // ignore filesystem classpath
+            // make sure it exists
+            File file = new File(el);
+
+            // A directory is a special case.
+            if (file.isDirectory()) {
+                // search for all .class files recursively starting with no prefix.
+                addClassFilesInTree(file, "");
                 continue;
             }
 
-            // make sure it exists
-            File file = new File(el);
             if (!file.exists() || !file.canRead())
                 continue;
```

Comment:
I think this check for file existence and readability should continue to be right after the File object is newed, and therefore before the file.isDirectory() check.